packages/wonder-blocks-data/util/request-handler.js

```diff
@@ -0,0 +1,40 @@
+// @flow
+import type {IRequestHandler} from "./types.js";
+
+/**
+ * Base implementation for creating a request handler.
+ *
+ * Provides a base implementation of the `IRequestHandler` base class for
+ * use with the Wonder Blocks Data framework.
+ */
+export default class RequestHandler<TOptions, TData>
+    implements IRequestHandler<TOptions, TData> {
+    _type: string;
+
+    constructor(type: string) {
+        this._type = type;
+    }
+
+    get type() {
+        return this._type;
+    }
+
+    invalidateCache(options: TOptions): boolean {
+        /**
+         * By default, the cache is always valid.
+         */
+        return false;
+    }
+
+    getKey(options: TOptions): string {
+        try {
+            return String(JSON.stringify(options));
```

Comment:
Why are you wrapping this with a `String(...)` - `JSON.stringify` already returns a string. If this is a Flow thing then just casting the value should be fine.

Reply:
`JSON.stringify` can return `undefined` which is not a valid string. We want an actual string here, always.

Reply:
It seems like this only happens if a string isn't passed in, like: `JSON.stringify(undefined)` or `JSON.stringify(function(){})`. I think in any case where `undefined` is returned we definitely want to treat that like an error and not just swallow it, converting it to a string.

Reply:
I changed the flow types for `JSON.stringify` in https://github.com/Khan/webapp/blob/master/flow-typed/json.js to get around this.  We could do the same here.

Reply:
I don't want to block `undefined`. It's a totally valid thing for options to be if, say, a handler doesn't even have options.

I could handle it differently though.

Reply:
I've replaced this use of `String` with a check for undefined and a cast.